hbase-client/src/main/java/org/apache/hadoop/hbase/client/LogQueryFilter.java

```diff
@@ -22,22 +22,27 @@
 import org.apache.commons.lang3.builder.EqualsBuilder;
 import org.apache.commons.lang3.builder.HashCodeBuilder;
 import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.hadoop.hbase.ServerName;
 import org.apache.yetus.audience.InterfaceAudience;
+import org.apache.yetus.audience.InterfaceStability;
+import java.util.Collections;
+import java.util.Set;
 
 /**
  * Slow/Large Log Query Filter with all filter and limit parameters
  * Used by Admin API: getSlowLogResponses
  */
-@InterfaceAudience.Private
-public class LogQueryFilter {
+@InterfaceAudience.Public
+@InterfaceStability.Evolving
+public class LogQueryFilter extends LogRequest {
 
   private String regionName;
   private String clientAddress;
   private String tableName;
   private String userName;
-  private int limit = 10;
```

Comment:
This change doesn't track with the javadoc for this class, which says "Slow/Large Log Query Filter with all filter and limit parameters". Also I think the comment should be updated to not mention the slow log too. 

Anyway, does it make sense to have the limit in the admin API but also here? Maybe a filter would want to filter more? Not a big deal, just wondering.

Reply:
This filter payload might be used to filter more as you said, you are right. But let me update javadoc to remove `limit` from description. Also, we don't need to remove mention of slow log because this filter is slowLog specific. Parent request payload, which is generic is `LogRequest` (which is empty so far but I have kept it IS.Evolving just in case we can identify some common filter param applicable to all use-cases).

Reply:
If we can limit the slow log result set size with this filter that is slow log specific, then we don't need a limit param on the existing admin API for slow log. So keep this?

Apologies for the back and forth on this.

Reply:
Brought back `limit` to request payload.

Reply:
-1 to limits on non public RPC request classes.

Reply:
In the future please do not mark unresolved conversations as resolved.

Reply:
> Limits by way of filter will be fine if you want to do it this way.

This is exactly what I am trying to do: 
1. For SlowLog use-case, this class exposes `limit`:
    `public class LogQueryFilter extends LogRequest`
2. For BalancerDecision use-case, this class exposes `limit`:
    `public class BalancerDecisionRequest extends LogRequest`